source/io/module_eggShell.js

```diff
@@ -239,6 +246,257 @@
             return result;
         };
 
+        Module.eggShell.typeTopAQSize = function (typePointer) {
+            return Module._TypeRef_TopAQSize(typePointer);
+        };
+
+        Module.eggShell.typeName = function (typePointer) {
+            return TypeRef_Name(typePointer);
+        };
+
+        Module.eggShell.typeSubElementCount = function (typePointer) {
+            return TypeRef_SubElementCount(typePointer);
+        };
+
+        Module.eggShell.typeSubElementByIndex = function (typePointer) {
+            return TypeRef_GetSubElementByIndex(typePointer);
+        };
+
+        Module.eggShell.typeIsCluster = function (typePointer) {
+            return Module._TypeRef_IsCluster(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsArray = function (typePointer) {
+            return Module._TypeRef_IsArray(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsBoolean = function (typePointer) {
+            return Module._TypeRef_IsBoolean(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsInteger = function (typePointer) {
+            return Module._TypeRef_IsInteger(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsSigned = function (typePointer) {
+            return Module._TypeRef_IsSigned(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsEnum = function (typePointer) {
+            return Module._TypeRef_IsEnum(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsFloat = function (typePointer) {
+            return Module._TypeRef_IsFloat(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsString = function (typePointer) {
+            return Module._TypeRef_IsString(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsPath = function (typePointer) {
+            return Module._TypeRef_IsPath(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsTimestamp = function (typePointer) {
+            return Module._TypeRef_IsTimestamp(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsComplex = function (typePointer) {
+            return Module._TypeRef_IsComplex(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsAnalogWaveform = function (typePointer) {
+            return Module._TypeRef_IsAnalogWaveform(typePointer) !== 0;
+        };
+
+        var validateVisitMethod = function (fn, fnName) {
```

Comment:
Consider renaming validateVisitMethod to validateVisitFunction.  Consider renaming fn to functionToVisit or methodToVisit depending on the name of the function you selected.